java/ql/src/Security/CWE/CWE-295/AndroidMissingCertificatePinning.qhelp

```diff
@@ -0,0 +1,48 @@
+<!DOCTYPE qhelp PUBLIC
+  "-//Semmle//qhelp//EN"
+  "qhelp.dtd">
+<qhelp>
+<overview>
+<p>
+Certificate pinning is the practice of only trusting a specific set of SSL certificates, rather than those that the device trusts by default.
+In Android applications, it is reccomended to use certificate pinning when communicating over the network, 
+in order to minimize the risk of machine-in-the-middle attacks from a comprimised CA.
+</p>
+</overview>
+
+<recommendation>
+<p>
+The easiest way to implement certificate pinning is to declare your pins in a <code>network-security-config</code> XML file. 
+This will automatically provide certificate pinning for any network connection made by the app.
+</p>
+<p>
+Another way to implement certificate pinning is to use the `CertificatePinner` from the `okhttp` library. 
```

Comment:
Minor comment. I'd remove the `the` before `CertificatePinner` OR leave it, but then add the name of the element type, such as ....."to use the `CertificatePinner` declaration/class/etc."

Sorry, not a developer, so I don't know what type of parameter `CertificatePinner` is 😄 (what I mean is to replace `declaration/class/etc.` with the correct term.

Reply:
```suggestion
Another way to implement certificate pinning is to use the `CertificatePinner` class from the `okhttp` library. 
```